Tests/test_image_array.py

```diff
@@ -27,6 +31,9 @@ def test(mode):
     assert test("RGBA") == ((100, 128, 4), "|u1", 51200)
     assert test("RGBX") == ((100, 128, 4), "|u1", 51200)
 
+    test_with_dtype(numpy.float)
```

Comment:
```suggestion
    test_with_dtype(float)
```
Running the test with `numpy.float`, I get
> DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.

Reply:
We could use numpy.float64 if that suppresses the warning, too.
The reason I would hesitate to use float is that I would have to research what the minimum version of numpy is that supports this.
But I'll follow your advice if you say that PIL is happy to just be compatible with recent numpy.

Reply:
Checking with Python 3.6 (the lowest version we support) and NumPy 1.12.0 (the lowest to support Python 3.6), this script runs fine:

```python
import numpy as np

np.array([1, 2, 3], dtype=np.float)
np.array([1, 2, 3], dtype=float)
print("ok")
```

NumPy 1.12.0 was released Jan 16, 2017, so I guess that's old enough.

And checking NumPy download stats at [pepy.tech](https://pepy.tech/project/numpy?versions=*&versions=1.21.*&versions=1.12.*&versions=1.11.*), those older versions account for a tiny proportion:

![image](https://user-images.githubusercontent.com/1324225/124125055-dc80d880-da81-11eb-878b-4e4dbb0d2332.png)